### PR TITLE
Update README.md with limitations in Windows

### PR DESCRIPTION
**Pull request by Chematronix**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ A full-featured REPL (interactive shell), with:
 * Auto hot-reload code anywhere, with state preserved
 * Supports full grammar in REPL
 * Play with existing code side-by-side
+* Support Linux and other *nixes (MacOS included?)
```
Comment (fzyzcjy) [inline README.md:18]: I am on MacOS and it works well

```suggestion
* Support Linux and other *nixes such as MacOS
```

**Comment by fzyzcjy** [comment]
Hmm, I wonder why Windows has those problems and whether we can avoid them? Theoretically speaking, this package does not use any *nix-specific api at all!

**Comment by Chematronix** [comment]
> Hmm, I wonder why Windows has those problems and whether we can avoid them? Theoretically speaking, this package does not use any *nix-specific api at all!

It's a dart:io bug (dart-lang/sdk#48329) which impacts cli_repl.  I'ts been mentioned before #81 [#88](https://github.com/fzyzcjy/dart_interactive/issues/88#issuecomment-1993770527) (and I see an unfinished PR at #61).  Fortunately it's easy to install WSL.

I would worry more about "Hot reload failed" on syntax/type errors, which completely breaks the REPL in all platforms and wipes all your progress.  At least that's what happens if I do something like `print()` or `1+'a'`.

**Comment by fzyzcjy** [comment]
I see. Then maybe we should explain it is dart:io bug. No worries just leave this pr as-is and I will update it.

> I would worry more about "Hot reload failed" on syntax/type errors, which completely breaks the REPL in all platforms and wipes all your progress. At least that's what happens if I do something like print() or 1+'a'.

I do not have windows at hand, so I wonder whether you mean "Hot reload failed" is also caused by this dart:io bug? Or by something else

**Comment by Chematronix** [comment]
> I do not have windows at hand, so I wonder whether you mean "Hot reload failed" is also caused by this dart:io bug? Or by something else

It happens in Windows and WSL Linux too, so I'd assume it breaks other platforms/OSes as well, if not all of them.  It's pretty aggravating.  This doesn't happen in MacOS?

image

**Comment by fzyzcjy** [comment]
Oh I see. Btw it would be great if we fix this, and feel free to PR for this!

**Comment by fzyzcjy** [comment]
(I will merge a bit later since will need to do a bit of modifications after merging)

**Comment by fzyzcjy** [comment]
@all-contributors please add @Chematronix for code

**Comment by fzyzcjy** [comment]
@all-contributors please add @Chematronix for doc

**Pull request merged**